src/config-context.ts

```diff
@@ -1,8 +1,25 @@
-import { createContext } from 'react'
+import { createContext, createElement, useContext, FC } from 'react'
 
 import { SWRConfiguration } from './types'
 
-const SWRConfigContext = createContext<SWRConfiguration>({})
-SWRConfigContext.displayName = 'SWRConfig'
+export const SWRConfigContext = createContext<SWRConfiguration>({})
 
-export default SWRConfigContext
+const SWRConfig: FC<{
+  value: SWRConfiguration
+}> = ({ children, value }) => {
+  // Extend middlewares of parent config context.
+  const parentConfigContext = useContext(SWRConfigContext)
+  if (parentConfigContext && parentConfigContext.middlewares) {
+    value = {
+      ...value,
```

Comment:
why only want to inherit the parent `middlewares` config instead of all of config fields?

Reply:
Fixed, good call!